CanonizerLoginPage.py

```diff
@@ -80,3 +83,169 @@ def login_with_valid_user(self, user, password):
         self.login(user, password)
 
         return CanonizerLoginPage(self.driver)
+
+    def login_page_mandatory_fields_are_marked_with_asterisk(self):
+        """
+        This Function checks, if Mandatory fields on Register Page Marked with *
+        First Name, Last Name, Email, Password, Confirm Password are Mandatory Fields
+
+        :return: the element value
+        """
+        return \
+            self.find_element(*LoginPageIdentifiers.EMAIL_ASTRK) and \
+            self.find_element(*LoginPageIdentifiers.PASSWORD_ASTRK)
+
+    def click_on_login_button(self):
+        self.hover(*LoginPageIdentifiers.LOGIN_BUTTON)
+        self.find_element(*LoginPageIdentifiers.LOGIN_BUTTON).click()
+
+        return CanonizerLoginPage(self.driver)
+
+    def click_on_close_icon_button(self):
+        self.click_on_login_button()
+        self.hover(*LoginPageIdentifiers.CLOSE_BUTTON)
+        self.find_element(*LoginPageIdentifiers.CLOSE_BUTTON).click()
+
+        return CanonizerLoginPage(self.driver)
+
+    def verify_the_login_with_invalid_email_format(self, default_invalid_user, default_pass):
+        self.click_on_login_button()
+        self.find_element(*LoginPageIdentifiers.EMAIL).send_keys(default_invalid_user)
+        self.find_element(*LoginPageIdentifiers.PASSWORD).send_keys(default_pass)
+        self.find_element(*LoginPageIdentifiers.SUBMIT).click()
+        self.find_element(*LoginPageIdentifiers.INVALID_EMAIL_TITLE).click()
+        title = self.find_element(*LoginPageIdentifiers.INVALID_EMAIL_TITLE).text
+        if title == message['Login_Page']['INVALID_EMAIL']:
+            return CanonizerLoginPage(self.driver)
+        else:
+            print("Error message not found")
+
+    def verify_the_login_with_blank_email(self, blank_email, default_pass):
+        self.click_on_login_button()
```

Comment:
Please create a function and remove the duplicate code. 124-129 repeated again in 185-190